app/code/Magento/NewsletterApi/Exception/AlreadySubscribedException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
```

Comment:
working further with the interfaces brought some new thoughts up for me.

I think the Newsletter Repository should throw the given internal Exceptions, as there are already plenty of good / speaking exceptions for the persistence layer itself.

the newsletter management on the other hand is not part of the persistence layer anymore and should have it's own exceptions.

If I delete a subscription entity over the repository, it is pretty clear what can happen and I can throw the default exceptions.

if I unsubscribe over the newsletter management, it is up to the integrator what happens then (e.g. delete entity or maybe just change subscription status to unsubscribed)

So my conclusion is, for the management, there should be specialized exceptions.